samcli/lib/intrinsic_resolver/intrinsic_property_resolver.py

```diff
@@ -492,7 +492,7 @@ def handle_find_in_map(self, intrinsic_value, ignore_errors):
         )
 
         second_level_value = top_level_value.get(second_level_key)
-        verify_intrinsic_type_str(
+        verify_non_null(
```

Comment:
This is changing the general intrinsic resolver functionality, which might have other side affects. Can you explain how you tested this change? Also, I couldn't find any tests to validate this change.

Reply:
Sorry I forgot to add the unit test, will do it now. Thanks!

Reply:
Thanks for adding those tests!

One final comment (at least for now) what happens if the key doesn't exist in the map? What would be the difference between previous and new behavior there?

Reply:
Thanks for the feedback!

There is no change with the behavior of `Fn:FindInMap` when the key doesn't exist in the map. The difference is when the value is not in String format, it won't return empty value anymore but instead will resolve into a value.